### ISSUE TITLE
[JENKINS-264] Mark projects a exclusive

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-06-a2#1734]
<p>I'd like to mark a project as "exclusive". No other project that is marked as<br/>
exclusive can be run at the same time.</p>

<p>This is useful for projects that require exclusive access to resources i.e.<br/>
network ports</p>

<p>Resolving this issue would allow me to build my target containers in parallel.<br/>
Tests however need to run sequetially.</p>

<p>cheers<br/>
-thomas</p>

---
<details><summary><i>Originally reported by <a title="Thomas Diesler" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tdiesler">tdiesler</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-264" target="_blank">Mark projects a exclusive</a></i></summary>
<i><ul>
<li><b>reported by</b>: tdiesler
<li><b>status</b>: Closed
<li><b>priority</b>: Major
<li><b>resolution</b>: Cannot Reproduce
<li><b>resolved</b>: 2009-03-07T09:07:52+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 0
<li><b>imported</b>: 2025-11-06
</ul></i>
</details>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kohsuke">kohsuke</a>:</i>
<p>I suspect this is a duplicate of issue #220. Is this not?</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tdiesler">tdiesler</a>:</i>
<p>I think issue #220 describs a differnt problem.</p>

<p>More generally, I want to say </p>

<p>Job1 extends TypeA<br/>
Job2 extends TypeA<br/>
Job3 extends TypeA</p>

<p>Only one instance of TypeA is allowed to run at any given type.<br/>
Providing an exclusive flag would solve the problem for me, but would not allow<br/>
for different job types.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kohsuke">kohsuke</a>:</i>
<p>Would latches&locks plugin help you?</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mindless">mindless</a>:</i>
<p>closing old issue (locks&latches can do this)</p>
